### ISSUE TITLE
When using empty arrays, Worksheets won't get added, resulting in a corrupt xlsx

### ISSUE THREAD

**Issue by moonwitch** [RamblingCookieMonster/PSExcel#48]
`Index operation failed; the array index evaluated to null.
At C:\Program Files\WindowsPowerShell\Modules\PSExcel\1.0.2\Export-XLSX.ps1:477 char:17
+ ...             $WorkSheet.Cells[$WorkSheet.Dimension.Address].AutoFitCol ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
    + FullyQualifiedErrorId : NullArrayIndex`

Long story short, I have to create a report where I have a list of AD groups, the report follows the format AD Groupname as Sheetname, the groupmembers go into the sheets. Some groups are empty, and on those groups - this errors out.

**Comment by moonwitch** [comment]
I would like to add: This is indeed caused by feeding $null into the width (-Autofit)
Not sure if this is a bug then, but if you use the switch and have empty values (PSObject), you will have a corrupt xlsx.

**Comment by RamblingCookieMonster** [comment]
Hi!

Good find, and thanks for the follow up detail, should make it easier to track down!

Feel free to poke around if you'd like, would be happy to take a PR - otherwise, will try to take a peek (might be a few weeks).

Cheers!

**Comment by RamblingCookieMonster** [comment]
Not published in the gallery, but put in a test to skip applying autofit if there's no data - guessing there might be other switches with similar behavior.  Thanks for pointing out where and why this was failing!
